app/controllers/provider_interface/references_controller.rb

```diff
@@ -1,10 +1,8 @@
 module ProviderInterface
   class ReferencesController < ProviderInterfaceController
-    before_action :set_application_choice
+    before_action :set_application_choice, :set_references
 
     def index
```

Comment:
I think we should redirect back to the main application tab here if the application is unsuccessful, so that providers can’t just see the references by editing the URL?

```suggestion
    def index
      return redirect_to(...) if application_choice.application_unsuccessful?
```

Reply:
Done!